core/dbt/contracts/graph/manifest.py

```diff
@@ -169,7 +170,33 @@ def find(
     ):
         unique_id = self.get_unique_id(key, package, version)
         if unique_id is not None:
-            return self.perform_lookup(unique_id, manifest)
+            node = self.perform_lookup(unique_id, manifest)
+            if version is None and node.is_versioned and get_node_info():
```

Comment:
what is `get_node_info()` doing here?

Reply:
This was my janky way of logging the message only at runtime, rather than at parsing time. We could log it twice.

Reply:
Ahhh I see! Worth a comment